src/OrleansProviders/Streams/Memory/MemoryBatchContainer.cs

```diff
@@ -60,5 +57,10 @@ public bool ShouldDeliver(IStreamIdentity stream, object filterData, StreamFilte
         {
             return true;
         }
+
+        void IOnDeserialized.OnDeserialized(ISerializerContext context)
+        {
+            this.serializer = ActivatorUtilities.GetServiceOrCreateInstance<TSerializer>(context.ServiceProvider);
```

Comment:
sounds good, although I'd make this field Lazy<ObjectFactory> so that exceptions flow to the right place, and not on type initialization.